### ISSUE TITLE
Analyzer to simplify LINQ expressions by using an overload that accepts a Func<TSource,Boolean>

### ISSUE THREAD

**Issue by Bartleby2718** [dotnet/roslyn#75844]
**Brief description:**

Some LINQ methods have an overload that accepts as an extra parameter `Func<TSource,Boolean>` to filter the result. We can simplify some LINQ expressions by taking advantage of such overloads.

**Languages applicable:**

Both C# and Visual Basic

**Code example that the analyzer should report:**

```cs
int[] numbers = [1, 2, 3, 4, 5, 6, 7];
// before
var oddNumberCount = numbers.Where(n => n % 2 == 1).Count();
// after
var oddNumberCount = numbers.Count(n => n % 2 == 1);
```

**Additional information:**

Relevant LINQ methods I can think of:
- `Count`,
- `First`
- `FirstOrDefault`
- `Last`
- `LastOrDefault`
- `Single`
- `SingleOrDefault`

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by davidwengier** [comment]
Isn't this https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0120 ?

**Comment by Bartleby2718** [comment]
@davidwengier It is! Closing as a duplicate.
